src/main/java/racingcar/controller/Game.java

```diff
@@ -0,0 +1,47 @@
+package racingcar.controller;
+
+import java.util.ArrayList;
+import java.util.List;
+import racingcar.common.type.TrialCount;
+import racingcar.domain.RacingCars;
+import racingcar.dto.output.AllRoundDTO;
+
+/**
+ * 레이싱 게임의 로직을 담당하는 클래스
+ */
+public class Game {
+    private final RacingCars racingCars;
+
+    // private 생성자
+    private Game(RacingCars racingCars) {
+        this.racingCars = racingCars;
+    }
+
+    // Game 객체를 생성하여 반환
+    public static Game of(RacingCars racingCars) {
+        return new Game(racingCars);
+    }
+
+    // 레이싱 게임을 실행하고 그 결과를 반환
+    public List<AllRoundDTO> play(TrialCount trialCount) {
+        List<AllRoundDTO> allRoundResults = new ArrayList<>();
+        int count = trialCount.getCount();
+
+        for (int i = 0; i < count; i++) {
+            moveCars();
+            allRoundResults.add(fetchRoundResult());
+        }
+
+        return allRoundResults;
+    }
```

Comment:
TrialCount를 객체를 통해 관리한다면 getCount()를 통해 값을 가져와서 반복문을 돌리는 것 보단, 이를 도메인으로 보고
```suggestion
    // 레이싱 게임을 실행하고 그 결과를 반환
    public List<AllRoundDTO> play(TrialCount trialCount) {
        List<AllRoundDTO> allRoundResults = new ArrayList<>();

        while (!trialCount.hasRemainingCount()) {
            moveCars();
            allRoundResults.add(fetchRoundResult());
            trialCount.decreaseCount();
        }

        return allRoundResults;
    }
```
이런 느낌으로 count 관리에 대한 책임을 위임하는 것은 어떨까요?

Reply:
이번에 vo를 처음써봤는데 더욱 도메인으로 보는 연습이 필요할 것 같습니다 ㅎㅎ